simbad2k/simbad2k.py

```diff
@@ -96,67 +103,120 @@ def __init__(self, query, scheme):
         self.scheme = scheme
 
     def _clean_result(self, result):
+        """
+        Converts the results from the MPC into a dictionary of floats.
+        Extracts the object's name from the query results and adds it to the dictionary.
+        """
         cleaned_result = {}
         for key in self.keys:
             try:
                 value = float(result[key])
             except (ValueError, TypeError):
                 value = None
             cleaned_result[key] = value
+        # Build object name from returned Data
+        if result['number']:
+            if result.get('name',None):
+                # If there is a number and a name, use both with the format "name (number)", otherwise just use number
+                cleaned_result['name'] = f"{result['name']} ({result['number']})"
+            else:
+                cleaned_result['name'] = f"{result['number']}"
+                if result.get('object_type', None):
+                    # Add comet object type if it exists
+                    cleaned_result['name'] += result['object_type']
+        else:
+            cleaned_result['name'] = result['designation']
         return cleaned_result
 
+    def get_primary_designation(self):
+        """
+        Submit the object's name to the MPC's query-identifier API to get the object's preferred primary and
+        preliminary designations.
+        In the case of multiple possible targets (usually happens for multiple objects with the same name),
+        try to disambiguate with the following criteria:
+            * Choose the first target with a 'permid' that could be converted into an INT if searching for an asteroid.
+            * Return the first target with a 'permid' if searching for a comet.
+            * If no 'permid' is found, query the MPC again using the first target with a preliminary designation.
+        """
+        response = requests.get("https://data.minorplanetcenter.net/api/query-identifier",
+                                data=self.query.replace("+", " "))
+        identifications = response.json()
+        if identifications['disambiguation_list']:
```

Comment:
`identifications.get('disambiguation_list')` for a safe get to avoid a KeyError if this doesn't exist?